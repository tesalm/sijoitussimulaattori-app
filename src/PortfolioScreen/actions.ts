import { Dispatch } from 'redux';

import { PortfolioApiRequest } from '../utils/api';
import { Portfolio } from './reducers';

export enum ActionType {
  RequestPortfolioBegin = '[Portfolio] Data Request',
  RequestPortfolioSuccess = '[Portfolio] Data Success',
  RequestPortfolioFailure = '[Portfolio] Data Failure',
}

export type PortfolioAction =
  | RequestPortfolioBegin
  | RequestPortfolioSuccess
  | RequestPortfolioFailure;

export class RequestPortfolioBegin {
  readonly type = ActionType.RequestPortfolioBegin;
  constructor() {
    return { type: this.type };
  }
}

export class RequestPortfolioSuccess {
  readonly type = ActionType.RequestPortfolioSuccess;
  constructor(public portfolio: Portfolio) {
    return { type: this.type, portfolio };
  }
}

export class RequestPortfolioFailure {
  readonly type = ActionType.RequestPortfolioFailure;
  constructor(public error: Error) {
    return { type: this.type, error };
  }
}

const getPortfolioData = (name: string) => async (
  dispatch: Dispatch<PortfolioAction>
) => {
  dispatch(new RequestPortfolioBegin());
  try {
    const data = await PortfolioApiRequest(name);
    dispatch(new RequestPortfolioSuccess(data));
  } catch (error) {
    dispatch(new RequestPortfolioFailure(error));
  }
};

export { getPortfolioData };
