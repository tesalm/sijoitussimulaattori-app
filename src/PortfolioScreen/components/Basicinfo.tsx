import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { t } from '../../assets/i18n';
import { RootState } from '../../redux/reducers';
import { Portfolio } from '../reducers';
import { stockStyles } from '../styles';

export interface BasicinfoProps {
  portfolio?: Portfolio;
  loading?: boolean;
  error?: Error;
}

export class Basicinfo extends React.Component<BasicinfoProps> {
  constructor(props: BasicinfoProps) {
    super(props);
  }
  revenueColor = (revenue: number): typeof stockStyles.revenueValueGreen => {
    return revenue >= 0
      ? stockStyles.revenueValueGreen
      : stockStyles.revenueValueRed;
  };

  formatValue = (value: number, currency: string): string => {
    if (currency == 'USD') {
      return value + ' $';
    } else if (currency == 'EUR') {
      return value + ' €';
    }
    return value + ' $';
  };

  render() {
    const { portfolio, loading, error } = this.props;
    if (loading == undefined && portfolio == undefined && error == undefined) {
      return (
        <View>
          <Text>vituiks meni</Text>
        </View>
      );
    }

    if (loading) {
      return (
        <View style={stockStyles.loading}>
          <ActivityIndicator size="large" />
        </View>
      );
    } else {
      if (portfolio == undefined) {
        //TODO: Format the error message to user
        if (error) {
          return <Text>Error! {error.message} </Text>;
        } else {
          return <Text>Error! asöfeh {portfolio} </Text>;
        }
      }
    }

    return (
      <View>
        <View style={stockStyles.basicinfo}>
          <Text style={stockStyles.titleStyle}>{t('PortfolioPage.Title')}</Text>
        </View>
        <View style={stockStyles.basicinfo}>
          <View style={stockStyles.basicinfoSmallerComp}>
            <Text style={stockStyles.valueHeader}>
              {t('PortfolioPage.MarketValue')}
            </Text>
            <Text style={stockStyles.value}>{portfolio.marketValue}</Text>
            <Text style={stockStyles.valueHeader}>
              {t('PortfolioPage.Cash')}
            </Text>
            <Text style={stockStyles.value}>{portfolio.cash} </Text>
          </View>
          <View style={stockStyles.basicinfoSmallerComp}>
            <Text style={stockStyles.valueHeaderMiddle}>
              {t('PortfolioPage.TotalValue')}
            </Text>
            <Text style={stockStyles.valueMiddle}>
              {portfolio.marketValue + portfolio.cash}
            </Text>
          </View>

          <View style={stockStyles.basicinfoMidComp}>
            <Text style={stockStyles.valueHeaderRightSide}>
              {t('PortfolioPage.Revenue')}
            </Text>
            <Text style={this.revenueColor(4.52)}>{portfolio.revenue}</Text>
          </View>
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  portfolio: state.singlePortfolio.portfolio,
  loading: state.singlePortfolio.loading,
  error: state.singlePortfolio.error,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators({}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Basicinfo);
