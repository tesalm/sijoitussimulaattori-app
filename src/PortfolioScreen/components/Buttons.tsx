import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';

import { generalStyles } from '../../App/generalStyles';
import { t } from '../../assets/i18n';
import Icon, { IconNames } from '../../general/icon';

export const EventsTransactions = (): JSX.Element => {
  return (
    // TODO: Make onPress for these items that go to buy or sell view.
    <TouchableHighlight style={generalStyles.button}>
      <View style={generalStyles.buttonView}>
        <View style={generalStyles.buttonLogoView}>
          <Icon iconName={IconNames.manage} iconHeight={24} iconWidth={24} />
          <Text style={generalStyles.buttonTitle}>
            {t('PortfolioPage.Events')}
          </Text>
        </View>
        <Icon iconName={IconNames.open} iconHeight={24} iconWidth={24} />
      </View>
    </TouchableHighlight>
  );
};

export const Manage = (): JSX.Element => {
  return (
    // TODO: Make onPress for these items that go to buy or sell view.
    <TouchableHighlight style={generalStyles.button}>
      <View style={generalStyles.buttonView}>
        <View style={generalStyles.buttonLogoView}>
          <Icon iconName={IconNames.manage} iconHeight={24} iconWidth={24} />
          <Text style={generalStyles.buttonTitle}>
            {t('PortfolioPage.Manage')}
          </Text>
        </View>
        <Icon iconName={IconNames.open} iconHeight={24} iconWidth={24} />
      </View>
    </TouchableHighlight>
  );
};
