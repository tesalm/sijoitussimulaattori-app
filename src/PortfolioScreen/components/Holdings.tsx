import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';

import { t } from '../../assets/i18n';
import Icon from '../../general/icon';
import { Stock } from '../../MarketScreen/reducer';
import {
  calculateTotalMarketValue,
  calculateTotalRevenue,
  calculateTotalRevenueProcent,
  countRevenuePercentage,
  formatCurrency,
  formatRevenue,
  formatRevenueCurrency,
  revenueColor,
  valueColor,
} from '../../util/stock';
import { Portfolio, PortfolioStock } from '../reducers';
import { portfolioStyles } from '../styles';

interface HoldingsProps {
  portfolio?: Portfolio;
  loading?: boolean;
  error?: Error;
  stocks?: Array<Stock>;
}

interface HoldingsState {
  activeSections: Array<number>;
}

export class Holdings extends React.Component<HoldingsProps, HoldingsState> {
  constructor(props: HoldingsProps, state: HoldingsState) {
    super(props);
    this.state = { activeSections: [] };
  }

  renderHeader = (
    section: PortfolioStock,
    index: number,
    isActive: boolean
  ) => {
    const iconName = isActive ? 'arrowUp' : 'arrowDown';

    return (
      <View style={portfolioStyles.holdingsSubLogoView}>
        <View style={portfolioStyles.holdingsSubTitleView}>
          <Text style={portfolioStyles.holdingsSubTitle}>{section.name}</Text>
        </View>
        <View style={portfolioStyles.holdingsArrowView}>
          <Icon iconName={iconName} iconHeight={24} iconWidth={24} />
        </View>
      </View>
    );
  };

  renderContent = (section: PortfolioStock) => {
    const { error, stocks } = this.props;
    if (stocks == undefined) {
      let errorMessage;
      if (error) {
        errorMessage = error.message + ' ';
      }
      // TODO: Muokkaa error-teksti käyttäjälle.
      return <Text>Error! {errorMessage} </Text>;
    }

    //find right stock from the array.
    const rightStock = stocks.find((stock) => stock.symbol == section.symbol);
    if (rightStock == undefined) {
      let errorMessage;
      if (error) {
        errorMessage = error.message + ' ';
      }
      // TODO: Muokkaa error-teksti käyttäjälle.
      return <Text>Error! {errorMessage} </Text>;
    }
    if (
      rightStock.stockInfo.intraday === undefined ||
      rightStock.stockInfo.stockMetadata === undefined ||
      rightStock.stockInfo.historyData == undefined
    ) {
      let errorMessage;
      if (rightStock.stockInfo.metaError) {
        errorMessage = rightStock.stockInfo.metaError.message + ' ';
      }
      if (rightStock.stockInfo.intraError) {
        errorMessage = errorMessage + rightStock.stockInfo.intraError.message;
      }
      if (rightStock.stockInfo.historyError) {
        errorMessage = errorMessage + rightStock.stockInfo.historyError.message;
      }
      // TODO: Muokkaa error-teksti käyttäjälle.
      return <Text>Error! {errorMessage} </Text>;
    }

    //calculate all necessary values needed.
    const currency = rightStock.stockInfo.stockMetadata.currency;
    const close = rightStock.stockInfo.intraday.intradayQuote[0].close;
    const totalRevenueProcent = calculateTotalRevenueProcent(
      rightStock,
      section
    );
    const totalMarketValue = calculateTotalMarketValue(rightStock, section);
    const totalRevenue = calculateTotalRevenue(rightStock, section);
    const revenueProcent = countRevenuePercentage(rightStock);

    return (
      <View>
        <View style={portfolioStyles.holdingsContainer}>
          <View style={portfolioStyles.portfolioInfoSmallerComp}>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.StocksOwned')}
            </Text>
            <Text style={portfolioStyles.value}>{section.amount}</Text>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.MarketValue')}
            </Text>
            <Text style={portfolioStyles.value}>
              {formatCurrency(close, currency)}{' '}
            </Text>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.TotalRevenueProcent')}
            </Text>
            <Text style={valueColor(totalRevenueProcent)}>
              {formatRevenue(totalRevenueProcent)}{' '}
            </Text>
          </View>
          <View style={portfolioStyles.portfolioInfoSmallerComp}>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.Acquisition')}
            </Text>
            <Text style={portfolioStyles.value}>
              {formatCurrency(section.avgPrice, currency)}
            </Text>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.TotalMarketValue')}
            </Text>
            <Text style={portfolioStyles.value}>
              {formatCurrency(totalMarketValue, currency)}{' '}
            </Text>
            <Text style={portfolioStyles.valueHeader}>
              {t('PortfolioPage.TotalRevenue')}
            </Text>
            <Text style={valueColor(totalRevenue)}>
              {formatRevenueCurrency(totalRevenue, currency)}{' '}
            </Text>
          </View>

          <View style={portfolioStyles.portfolioInfoSmallerComp}>
            <Text style={portfolioStyles.valueHeaderRightSideHoldings}>
              {t('PortfolioPage.Revenue')}
            </Text>
            <Text style={revenueColor(revenueProcent)}>
              {formatRevenue(revenueProcent)}
            </Text>
          </View>
        </View>
      </View>
    );
  };

  updateSections = (activeSections: number[]) => {
    this.setState({ activeSections });
  };

  render() {
    const { portfolio, error, loading, stocks } = this.props;
    if (loading) {
      return (
        <View style={portfolioStyles.loading}>
          <ActivityIndicator size="large" />
        </View>
      );
    } else {
      if (
        portfolio === undefined ||
        stocks == undefined ||
        portfolio == undefined
      ) {
        let errorMessage;
        if (error) {
          errorMessage = error.message + ' ';
        }
        // TODO: Muokkaa error-teksti käyttäjälle.
        return <Text>Error! {errorMessage} </Text>;
      }
    }

    return (
      <View>
        <View style={portfolioStyles.holdingsLogoView}>
          <Icon iconName={'holdings'} iconHeight={24} iconWidth={24} />
          <Text style={portfolioStyles.titleText}>
            {t('PortfolioPage.Holdings')}
          </Text>
        </View>
        <Accordion
          expandMultiple={true}
          sections={portfolio.stocks}
          activeSections={this.state.activeSections}
          renderHeader={this.renderHeader}
          renderContent={this.renderContent}
          onChange={this.updateSections}
          sectionContainerStyle={portfolioStyles.accordionContainer}
        />
      </View>
    );
  }
}
