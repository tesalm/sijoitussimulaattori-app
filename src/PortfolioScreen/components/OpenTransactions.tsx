import React from 'react';
import { ActivityIndicator, Alert, Text, TouchableOpacity, View } from 'react-native';
import Collapsible from 'react-native-collapsible';
import { scale, verticalScale } from 'react-native-size-matters';

import { t } from '../../assets/i18n';
import Icon from '../../general/icon';
import { cancelTransaction, getTransactions } from '../../PortfolioList/actions';
import { Transaction } from '../../PortfolioList/reducer';
import { portfolioStyles } from '../styles';

export interface TransactionProps {
  transactions: Array<Transaction>;
  loading: boolean;
  error?: Error;
  portfolioId?: string;
  cancelOpenTransaction: typeof cancelTransaction;
  getTransactions: typeof getTransactions;
}

interface TransactionState {
  collapsed: boolean;
  fetched: boolean;
}

export class OpenTransactions extends React.Component<
  TransactionProps,
  TransactionState
> {
  constructor(props: TransactionProps) {
    super(props);
    this.state = { collapsed: true, fetched: false };
  }

  toggleExpanded = () => {
    this.setState({ collapsed: !this.state.collapsed });
    if (!this.state.fetched && this.props.portfolioId) {
      this.props.getTransactions(this.props.portfolioId);
      this.setState({ fetched: true });
    }
  };

  renderToggleButton = () => {
    const iconName = this.state.collapsed ? 'arrowDown' : 'arrowUp';
    return (
      <View style={{ flex: 1, flexDirection: 'row' }}>
        <Icon iconName={'opensign'} iconHeight={24} iconWidth={24} />
        <View style={{ flex: 1, flexDirection: 'row' }}>
          <Text style={portfolioStyles.titleText}>
            {t('OpenTransactions.ToggleTitle')}
          </Text>
          {this.props.loading && (
            <ActivityIndicator
              size={verticalScale(24)}
              style={{ marginLeft: scale(18) }}
            />
          )}
        </View>
        <Icon iconName={iconName} iconHeight={24} iconWidth={24} />
      </View>
    );
  };

  renderContent = () => {
    const { error, transactions, portfolioId } = this.props;
    if (error) {
      return (
        <Text style={portfolioStyles.noActionsText}>
          Error! {error.message}
        </Text>
      );
    }
    if (transactions.length < 1 || portfolioId === undefined) {
      return (
        <Text style={portfolioStyles.noActionsText}>
          {t('OpenTransactions.NoTransactions')}
        </Text>
      );
    }

    return transactions.map((item, i) => (
      <View key={i} style={portfolioStyles.transaction}>
        <Text style={portfolioStyles.symbol}>{item.symbol}</Text>
        <View style={portfolioStyles.portfolioinfo}>
          <View style={{ flex: 1 }}>
            <Text style={portfolioStyles.basicText}>
              {t('OpenTransactions.Amount')}
            </Text>
            <Text style={portfolioStyles.value}>{item.amount}</Text>
          </View>
          <View style={{ flex: 1.5 }}>
            <Text style={portfolioStyles.basicText}>
              {t('OpenTransactions.Total')}
            </Text>
            <Text style={portfolioStyles.value}>{item.price} â‚¬</Text>
          </View>
          <TouchableOpacity
            onPress={() =>
              Alert.alert(
                t('OpenTransactions.AlertTitle'),
                t('OpenTransactions.ConfTxtP1') +
                  item.symbol +
                  t('OpenTransactions.ConfTxtP2'),
                [
                  {
                    text: t('OpenTransactions.N'),
                    style: 'cancel',
                  },
                  {
                    text: t('OpenTransactions.Y'),
                    onPress: () =>
                      this.props.cancelOpenTransaction(portfolioId, item.uid),
                  },
                ]
              )
            }
          >
            <Text style={portfolioStyles.cancelAction}>
              {t('OpenTransactions.CancelAction')}
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    ));
  };

  render() {
    return (
      <View>
        <TouchableOpacity onPress={this.toggleExpanded}>
          {this.renderToggleButton()}
        </TouchableOpacity>
        {!this.props.loading && (
          <Collapsible collapsed={this.state.collapsed}>
            {this.renderContent()}
          </Collapsible>
        )}
      </View>
    );
  }
}
