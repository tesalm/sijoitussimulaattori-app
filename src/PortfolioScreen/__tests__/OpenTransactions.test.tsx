import { shallow } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';

import { cancelTransaction } from '../../PortfolioList/actions';
import * as actions from '../../PortfolioList/actions';
import { PortfolioListing, portfolioListingReducer } from '../../PortfolioList/reducer';
import { OpenTransactions, TransactionProps } from '../components/OpenTransactions';

import 'react-native';

describe('Portfolio pending transactions tests: ', () => {
  const middlewares: any = [];
  const mockStore = configureStore(middlewares);

  const defaultTransactionProps: TransactionProps = {
    transactions: [],
    loading: false,
    error: undefined,
    portfolioId: 'Portfolio1',
    getTransactions: jest.fn(),
    cancelOpenTransaction: jest.fn(),
  };

  const successTransactionProps: TransactionProps = {
    transactions: [
      {
        uid: '1',
        type: 'SELL',
        symbol: 'AMZN',
        amount: 20,
        price: 222.2,
        expiresAt: '2018-12-14T15:55:00.000Z',
        state: 'MARKET',
        fulfilledAt: '',
        cancelledAt: '',
      },
    ],
    loading: false,
    error: undefined,
    portfolioId: 'Portfolio1',
    getTransactions: jest.fn(),
    cancelOpenTransaction: cancelTransaction,
  };

  const errorTransactionProps: TransactionProps = {
    transactions: [],
    loading: false,
    error: {
      name: 'Network Error',
      message: 'Network connection failed',
    },
    portfolioId: 'Portfolio1',
    getTransactions: jest.fn(),
    cancelOpenTransaction: jest.fn(),
  };

  const initialState: PortfolioListing = {
    portfolioListing: [
      {
        uid: 'Portfolio1',
        name: 'Portfolio1',
        balance: 500,
        totalRevenue: -0.02,
        totalMarketValue: 5000,
        lastDayRevenue: 0,
        ownerId: 'JohnDoe',
        portfolioInfo: {
          loading: false,
          error: undefined,
          refreshing: false,
          portfolio: {
            uid: 'Portfolio1',
            name: 'Portfolio1',
            balance: 500,
            totalRevenue: -0.02,
            totalMarketValue: 5000,
            lastDayRevenue: 0,
            ownerId: 'JohnDoe',
            stocks: [],
          },
        },
        transactions: {
          transactions: successTransactionProps.transactions,
          loading: false,
          error: undefined,
        },
      },
    ],
    loading: false,
    error: undefined,
    creatingPortfolio: {
      creatingPortfolioLoading: false,
      creatingPortfolioError: undefined,
      creatingPortfolioSuccess: false,
    },
  };

  it('renders correctly', async () => {
    const component = shallow(
      <OpenTransactions {...defaultTransactionProps} />,
      { context: { store: mockStore() } }
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with error', async () => {
    const component = shallow(<OpenTransactions {...errorTransactionProps} />, {
      context: { store: mockStore() },
    });
    expect(component.find('Collapsible')).toMatchSnapshot();
  });

  it('renders correctly with data', async () => {
    const component = shallow(
      <OpenTransactions {...successTransactionProps} />,
      { context: { store: mockStore() } }
    );
    expect(component.find('Collapsible')).toMatchSnapshot();
  });

  it('actions', async () => {
    expect(
      new actions.RequestTransactionsBegin('Portfolio1')
    ).toMatchSnapshot();
    expect(
      new actions.RequestTransactionsSuccess(
        successTransactionProps.transactions,
        'Portfolio1'
      )
    ).toMatchSnapshot();
    expect(
      new actions.RequestCancelTransactionSuccess('1', 'Portfolio1')
    ).toMatchSnapshot();

    const err: any = errorTransactionProps.error;
    expect(
      new actions.RequestTransactionsFailure(err, 'Portfolio1')
    ).toMatchSnapshot();
  });

  it('reducers', async () => {
    let action: any;
    action = {
      type: actions.ActionType.RequestTransactionsBegin,
      portfolioId: 'Portfolio1',
    };
    expect(portfolioListingReducer(initialState, action)).toMatchSnapshot();
    action = {
      type: actions.ActionType.RequestTransactionsSuccess,
      transactions: successTransactionProps.transactions,
      portfolioId: 'Portfolio1',
    };
    expect(portfolioListingReducer(initialState, action)).toMatchSnapshot();
    action = {
      type: actions.ActionType.RequestTransactionsFailure,
      error: errorTransactionProps.error,
      portfolioId: 'Portfolio1',
    };
    expect(portfolioListingReducer(initialState, action)).toMatchSnapshot();
    action = {
      type: actions.ActionType.RequestCancelTransactionSuccess,
      tid: '1',
      portfolioId: 'Portfolio1',
    };
    expect(portfolioListingReducer(initialState, action)).toMatchSnapshot();
  });
});
