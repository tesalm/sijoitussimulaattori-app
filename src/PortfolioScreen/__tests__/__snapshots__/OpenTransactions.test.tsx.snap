// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio pending transactions tests:  actions 1`] = `
Object {
  "portfolioId": "Portfolio1",
  "type": "[Transactions] API Request",
}
`;

exports[`Portfolio pending transactions tests:  actions 2`] = `
Object {
  "portfolioId": "Portfolio1",
  "transactions": Array [
    Object {
      "amount": 20,
      "cancelledAt": "",
      "expiresAt": "2018-12-14T15:55:00.000Z",
      "fulfilledAt": "",
      "price": 222.2,
      "status": "MARKET",
      "symbol": "AMZN",
      "type": "SELL",
      "uid": "1",
    },
  ],
  "type": "[Transactions] API Success",
}
`;

exports[`Portfolio pending transactions tests:  actions 3`] = `
Object {
  "portfolioId": "Portfolio1",
  "transactionId": "1",
  "type": "[Transactions] Cancel success",
}
`;

exports[`Portfolio pending transactions tests:  actions 4`] = `
Object {
  "error": Object {
    "message": "Network connection failed",
    "name": "Network Error",
  },
  "portfolioId": "Portfolio1",
  "type": "[Transactions] API Failure",
}
`;

exports[`Portfolio pending transactions tests:  reducers 1`] = `
Object {
  "creatingPortfolio": Object {
    "creatingPortfolioError": undefined,
    "creatingPortfolioLoading": false,
    "creatingPortfolioSuccess": false,
  },
  "error": undefined,
  "loading": false,
  "portfolioListing": Array [
    Object {
      "balance": 500,
      "lastDayRevenue": 0,
      "name": "Portfolio1",
      "ownerId": "JohnDoe",
      "portfolioInfo": Object {
        "error": undefined,
        "loading": false,
        "portfolio": Object {
          "balance": 500,
          "lastDayRevenue": 0,
          "name": "Portfolio1",
          "ownerId": "JohnDoe",
          "stocks": Array [],
          "totalMarketValue": 5000,
          "totalRevenue": -0.02,
          "uid": "Portfolio1",
        },
        "refreshing": false,
      },
      "totalMarketValue": 5000,
      "totalRevenue": -0.02,
      "transactionInfo": Object {
        "transactionSuccess": false,
        "transactions": undefined,
        "transactionsError": undefined,
        "transactionsLoading": false,
      },
      "transactions": Object {
        "error": undefined,
        "loading": true,
        "transactionListing": Array [
          Object {
            "amount": 20,
            "cancelledAt": "",
            "expiresAt": "2018-12-14T15:55:00.000Z",
            "fulfilledAt": "",
            "price": 222.2,
            "status": "MARKET",
            "symbol": "AMZN",
            "type": "SELL",
            "uid": "1",
          },
        ],
      },
      "uid": "Portfolio1",
    },
  ],
}
`;

exports[`Portfolio pending transactions tests:  reducers 2`] = `
Object {
  "creatingPortfolio": Object {
    "creatingPortfolioError": undefined,
    "creatingPortfolioLoading": false,
    "creatingPortfolioSuccess": false,
  },
  "error": undefined,
  "loading": false,
  "portfolioListing": Array [
    Object {
      "balance": 500,
      "lastDayRevenue": 0,
      "name": "Portfolio1",
      "ownerId": "JohnDoe",
      "portfolioInfo": Object {
        "error": undefined,
        "loading": false,
        "portfolio": Object {
          "balance": 500,
          "lastDayRevenue": 0,
          "name": "Portfolio1",
          "ownerId": "JohnDoe",
          "stocks": Array [],
          "totalMarketValue": 5000,
          "totalRevenue": -0.02,
          "uid": "Portfolio1",
        },
        "refreshing": false,
      },
      "totalMarketValue": 5000,
      "totalRevenue": -0.02,
      "transactionInfo": Object {
        "transactionSuccess": false,
        "transactions": undefined,
        "transactionsError": undefined,
        "transactionsLoading": false,
      },
      "transactions": Object {
        "error": undefined,
        "loading": false,
        "transactionListing": Array [
          Object {
            "amount": 20,
            "cancelledAt": "",
            "expiresAt": "2018-12-14T15:55:00.000Z",
            "fulfilledAt": "",
            "price": 222.2,
            "status": "MARKET",
            "symbol": "AMZN",
            "type": "SELL",
            "uid": "1",
          },
        ],
      },
      "uid": "Portfolio1",
    },
  ],
}
`;

exports[`Portfolio pending transactions tests:  reducers 3`] = `
Object {
  "creatingPortfolio": Object {
    "creatingPortfolioError": undefined,
    "creatingPortfolioLoading": false,
    "creatingPortfolioSuccess": false,
  },
  "error": undefined,
  "loading": false,
  "portfolioListing": Array [
    Object {
      "balance": 500,
      "lastDayRevenue": 0,
      "name": "Portfolio1",
      "ownerId": "JohnDoe",
      "portfolioInfo": Object {
        "error": undefined,
        "loading": false,
        "portfolio": Object {
          "balance": 500,
          "lastDayRevenue": 0,
          "name": "Portfolio1",
          "ownerId": "JohnDoe",
          "stocks": Array [],
          "totalMarketValue": 5000,
          "totalRevenue": -0.02,
          "uid": "Portfolio1",
        },
        "refreshing": false,
      },
      "totalMarketValue": 5000,
      "totalRevenue": -0.02,
      "transactionInfo": Object {
        "transactionSuccess": false,
        "transactions": undefined,
        "transactionsError": undefined,
        "transactionsLoading": false,
      },
      "transactions": Object {
        "error": Object {
          "message": "Network connection failed",
          "name": "Network Error",
        },
        "loading": false,
        "transactionListing": Array [
          Object {
            "amount": 20,
            "cancelledAt": "",
            "expiresAt": "2018-12-14T15:55:00.000Z",
            "fulfilledAt": "",
            "price": 222.2,
            "status": "MARKET",
            "symbol": "AMZN",
            "type": "SELL",
            "uid": "1",
          },
        ],
      },
      "uid": "Portfolio1",
    },
  ],
}
`;

exports[`Portfolio pending transactions tests:  reducers 4`] = `
Object {
  "creatingPortfolio": Object {
    "creatingPortfolioError": undefined,
    "creatingPortfolioLoading": false,
    "creatingPortfolioSuccess": false,
  },
  "error": undefined,
  "loading": false,
  "portfolioListing": Array [
    Object {
      "balance": 500,
      "lastDayRevenue": 0,
      "name": "Portfolio1",
      "ownerId": "JohnDoe",
      "portfolioInfo": Object {
        "error": undefined,
        "loading": false,
        "portfolio": Object {
          "balance": 500,
          "lastDayRevenue": 0,
          "name": "Portfolio1",
          "ownerId": "JohnDoe",
          "stocks": Array [],
          "totalMarketValue": 5000,
          "totalRevenue": -0.02,
          "uid": "Portfolio1",
        },
        "refreshing": false,
      },
      "totalMarketValue": 5000,
      "totalRevenue": -0.02,
      "transactionInfo": Object {
        "transactionSuccess": false,
        "transactions": undefined,
        "transactionsError": undefined,
        "transactionsLoading": false,
      },
      "transactions": Object {
        "error": undefined,
        "loading": false,
        "transactionListing": Array [],
      },
      "uid": "Portfolio1",
    },
  ],
}
`;

exports[`Portfolio pending transactions tests:  renders correctly 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Icon
        iconHeight={24}
        iconName="opensign"
        iconWidth={24}
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#004D40",
              "fontSize": 33.35,
              "fontWeight": "bold",
              "marginLeft": 17.142857142857142,
            }
          }
        >
          Open Transactions
        </Text>
      </View>
      <Icon
        iconHeight={24}
        iconName="arrowDown"
        iconWidth={24}
      />
    </View>
  </TouchableOpacity>
  <Collapsible
    align="top"
    collapsed={true}
    collapsedHeight={0}
    duration={300}
    easing="easeOutCubic"
    enablePointerEvents={false}
    onAnimationEnd={[Function]}
    style={
      Object {
        "marginTop": 11.770588235294117,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Roboto",
          "fontSize": 27.46470588235294,
          "marginBottom": 7.847058823529411,
          "marginTop": 29.426470588235293,
          "opacity": 0.6,
        }
      }
    >
      No pending transactions
    </Text>
  </Collapsible>
</View>
`;

exports[`Portfolio pending transactions tests:  renders correctly with data 1`] = `
<Collapsible
  align="top"
  collapsed={true}
  collapsedHeight={0}
  duration={300}
  easing="easeOutCubic"
  enablePointerEvents={false}
  onAnimationEnd={[Function]}
  style={
    Object {
      "marginTop": 11.770588235294117,
    }
  }
>
  <View
    key="0"
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 3.9235294117647057,
        "marginTop": 23.541176470588233,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontFamily": "Roboto",
            "fontSize": 27.46470588235294,
            "fontWeight": "bold",
            "marginBottom": 7.847058823529411,
            "textAlign": "justify",
          }
        }
      >
        SELL
      </Text>
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontFamily": "Roboto",
            "fontSize": 23.541176470588233,
            "textAlign": "justify",
          }
        }
      >
        Amount
      </Text>
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontSize": 27.46470588235294,
            "fontWeight": "bold",
            "marginBottom": 7.847058823529411,
            "marginTop": -3.9235294117647057,
            "textAlign": "justify",
          }
        }
      >
        20
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontFamily": "Roboto",
            "fontSize": 27.46470588235294,
            "fontWeight": "bold",
            "marginBottom": 7.847058823529411,
            "textAlign": "justify",
          }
        }
      >
        AMZN
      </Text>
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontFamily": "Roboto",
            "fontSize": 23.541176470588233,
            "textAlign": "justify",
          }
        }
      >
        Total cost
      </Text>
      <Text
        style={
          Object {
            "color": "#004D40",
            "fontSize": 27.46470588235294,
            "fontWeight": "bold",
            "marginBottom": 7.847058823529411,
            "marginTop": -3.9235294117647057,
            "textAlign": "justify",
          }
        }
      >
        222.2
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "orange",
            "fontFamily": "Roboto",
            "fontSize": 29.426470588235293,
          }
        }
      >
        CANCEL
      </Text>
    </TouchableOpacity>
  </View>
</Collapsible>
`;

exports[`Portfolio pending transactions tests:  renders correctly with error 1`] = `
<Collapsible
  align="top"
  collapsed={true}
  collapsedHeight={0}
  duration={300}
  easing="easeOutCubic"
  enablePointerEvents={false}
  onAnimationEnd={[Function]}
  style={
    Object {
      "marginTop": 11.770588235294117,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Roboto",
        "fontSize": 27.46470588235294,
        "marginBottom": 7.847058823529411,
        "marginTop": 29.426470588235293,
        "opacity": 0.6,
      }
    }
  >
    Error! 
    Network connection failed
  </Text>
</Collapsible>
`;
